### PR TITLE
Create controlplane API to enable HA CloudAgent without FailOver Cluster

### PR DESCRIPTION
**Pull request by bfjelds**
This PR is in support of https://github.com/microsoft/wssdcloudagent/pull/349

[rebased onto v0.10.5 on 12/4 ... but force push didn't do anything because the HEAD commit was 10/22 and had already been rebased to]
[rebased onto v0.10.6-alpha.2 on 12/14]

### CODE REVIEW

File: services/cloud/cloud.go [hunk 1]
```diff
@@ -248,3 +248,34 @@ type Cluster struct {
...
+
+// ControlPlaneProperties the resource group properties.
+type ControlPlaneProperties struct {
+	// State - State
+	Statuses map[string]*string `json:"statuses"`
+	// Fqdn
+	Fqdn *string `json:"fqdn,omitempty"`
+	// Port
+	Port *int32 `json:"port,omitempty"`
+	// AuthorizerPort
```
Comment (jiria) [inline services/cloud/cloud.go:260]: Can you please make these comments more descriptive?

File: services/cloud/cloud.go [hunk 2]
```diff
@@ -248,3 +248,34 @@ type Cluster struct {
...
+	autorest.Response `json:"-"`
+	// ID - READ-ONLY; The ID of the resource group.
+	ID *string `json:"id,omitempty"`
+	// Name - The name of the resource group.
+	Name *string `json:"name,omitempty"`
+	//Properties
+	*ControlPlaneProperties `json:"properties,omitempty"`
+	// Version
+	Version *string `json:"version,omitempty"`
+	// Location - The location of the resource group. It cannot be changed after the resource group has been created. It must be one of the supported Azure locations.
```
Comment (jiria) [inline services/cloud/cloud.go:277]: Is this really a local of a resource group?

File: services/cloud/controlplane/controlplane.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	if nd.Fqdn == nil {
+		return nil, errors.Wrapf(errors.InvalidConfiguration, "Missing FQDN in Configuration")
+	}
+
+	if nd.Port == nil {
+		return nil, errors.Wrapf(errors.InvalidConfiguration, "Missing Port in Configuration")
+	}
+
+	if nd.AuthorizerPort == nil {
+		return nil, errors.Wrapf(errors.InvalidConfiguration, "Missing AuthorizrPort in Configuration")
```
Comment (jiria) [inline services/cloud/controlplane/controlplane.go:29]: Nit: typo AuthorizerPort

File: services/cloud/controlplane/wssd.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+func (c *client) getControlPlaneFromResponse(response *wssdcloud.ControlPlaneResponse) *[]cloud.ControlPlaneInfo {
+	gps := []cloud.ControlPlaneInfo{}
+	for _, gp := range response.GetControlPlanes() {
+		gps = append(gps, *(getControlPlane(gp)))
+	}
+
+	return &gps
+}
+
+func (c *client) getControlPlaneRequest(opType wssdcloudcommon.Operation, location, name string, gpss *cloud.ControlPlaneInfo) (*wssdcloud.ControlPlaneRequest, error) {
```
Comment (jiria) [inline services/cloud/controlplane/wssd.go:136]: Why is gpss gpss?

File: services/compute/virtualmachineimage/virtualmachineimage.go [hunk 5]
```diff
@@ -31,8 +31,8 @@ func getWssdVirtualMachineImage(c *compute.VirtualMachineImage, groupName string
 // Conversion function from wssdcloudcompute to compute
 func getVirtualMachineImage(c *wssdcloudcompute.VirtualMachineImage, group string) *compute.VirtualMachineImage {
 	return &compute.VirtualMachineImage{
-		Name: &c.Name,
-		ID:   &c.Id,
+		Name:                          &c.Name,
```
Comment (jiria) [inline services/compute/virtualmachineimage/virtualmachineimage.go:34]: Unintentional?

File: services/cloud/controlplane/wssd.go (hunk 4 above)
Comment (bfjelds) [reply services/cloud/controlplane/wssd.go:132]: copy/paste of a copy/paste ... i think the original was for groups maybe?  will update here.

File: services/cloud/controlplane/controlplane.go (hunk 3 above)
Comment (bfjelds) [reply services/cloud/controlplane/controlplane.go]: removed authorizerPort

File: services/compute/virtualmachineimage/virtualmachineimage.go (hunk 5 above)
Comment (bfjelds) [reply services/compute/virtualmachineimage/virtualmachineimage.go:34]: yes

File: services/cloud/cloud.go (hunk 2 above)
Comment (bfjelds) [reply services/cloud/cloud.go:277]: no, it is the location of the controlplane .. .heh ... another copy/paste of a copy/paste (like all of the other location-based entitities in this file: Node, Kubernetes, & Cluster)

Comment (bfjelds) [reply services/cloud/cloud.go:277]: will fix.

File: services/cloud/cloud.go (hunk 1 above)
Comment (bfjelds) [reply services/cloud/cloud.go]: will do.

File: services/cloud/cloud.go (hunk 2 above)
Comment (madhanrm) [reply services/cloud/cloud.go:276]: can you rename ControlPlaneInfo to ControlPlane ?

**Comment by madhanrm** [comment]
everything looks good. Just left one comment

File: services/cloud/cloud.go (hunk 2 above)
Comment (bfjelds) [reply services/cloud/cloud.go:276]: the reason it is ControlPlaneInfo is ControlPlane conflicts with this:

```go
type NodeType string

const (
	ControlPlane  NodeType = "ControlPlane" // <---------- ControlPlane conflict
	LinuxWorker   NodeType = "LinuxWorker"
	WindowsWorker NodeType = "WindowsWorker"
	LoadBalancer  NodeType = "LoadBalancer"
)
```

**Review by madhanrm — APPROVED** [review]

**Pull request merged**